pgjdbc/src/main/java/org/postgresql/core/VisibleBufferedInputStream.java

```diff
@@ -146,10 +148,17 @@ private boolean readMore(int wanted) throws IOException {
   }
 
   /**
-   * Doubles the size of the buffer.
+   * Doubles the size of the buffer, guarantees it will be big enough to fit at least wanted bytes
+   * @param wanted minimum size required
    */
-  private void doubleBuffer() {
-    byte[] buf = new byte[buffer.length * 2];
+  private void doubleBuffer(int wanted) {
+
+    int bufferLength = buffer.length * 2;
```

Comment:
how about:

```
//find power of 2 less than wanted (i.e. highest bit) and then shift left 1 bit
final int bufferLength = Integer.highestOneBit(wanted) << 1;
final byte[] buf = new byte[bufferLength];
```

Reply:
I expect the JVM optimizes this.

Reply:
I am suggesting a different approach, not just bit twiddling instead of multiplication (which I agree likely gets optimized away). I am proposing driving off of `wanted` rather than doubling `buffer.length` in loop until it is larger than `wanted`.

Reply:
makes sense